### PR TITLE
DeNet File Token (DE)

### PR DESCRIPTION
**Pull request by xasai**
##### Name (to be shown on DefiLlama):
DeNet File Token 

##### Twitter Link:
https://twitter.com/denetpro

##### List of audit links if any:
https://github.com/pessimistic-io/audits/blob/a2acb36c7f4b2a61c579a29133bb2eab7e9ad894/DeNet%20Security%20Analysis%20by%20Pessimistic.pdf

##### Website Link:
https://denet.pro

##### Logo (High resolution, will be shown with rounded borders):
DE

##### Current TVL:
182.14 k

##### Treasury Addresses (if the protocol has treasury)
0x1a9b54A3075119f1546C52cA0940551A6ce5d2D0

##### Chain:
Polygon (chain_id 137)

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
denet-file-token

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
s-UH3iRSb+denetfiletoken-de

##### Short Description (to be shown on DefiLlama):
DeNet File Token (DE) serves as a means for users to open TBY storage deposits for storing data, as well as for Datakeepers to run nodes.
Mobile storage - https://links.denet.app/mobile \
Datakeeper node - https://p.denet.app/datakeeper

##### Token address and ticker if any:
0x081Ec4c0e30159C8259BAD8F4887f83010a681DC - DE 

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Services

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
N/A

##### forkedFrom (Does your project originate from another project):
N/A

##### methodology (what is being counted as tvl, how is tvl being calculated):
1. Summarizing the DE balances of the most popular liquidity pools. 
2. Adding up the total amount of DE tokens in the Payments contract used for opening TBY deposits.

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/denetpro

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/DeNet exports TVL: 
        
 
 
    polygon                   13.33 k
    
    total                    13.33 k

File: projects/DeNet/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const { sumTokensExport } = require("../helper/unwrapLPs");
+const DE_TOKEN_ADDRESS = '0x081Ec4c0e30159C8259BAD8F4887f83010a681DC';
+
+const owners = [
+  '0x1a9b54A3075119f1546C52cA0940551A6ce5d2D0', // deposit of TBY in DeNet Payments
```
[the remark this answers is not in the document]
Comment (g1nt0ki) [reply projects/DeNet/index.js]: hi @djdeniro 

hmm, we can count DE token value in both contracts under staking

    const { sumTokensExport } = require("../helper/unwrapLPs");
    const DE_TOKEN_ADDRESS = '0x081Ec4c0e30159C8259BAD8F4887f83010a681DC';

    const owners = [
      '0x1a9b54A3075119f1546C52cA0940551A6ce5d2D0', // veDe
    ]

    module.exports = {
      start: 1691761595, // Friday, 11-Aug-23 13:46:35 UTC
      methodology: "Total amount of DE tokens used for DeNet storage payments",
      polygon: {
        tvl: () => ({}),
        staking: sumTokensExport({
          owners: owners,
          tokens: [DE_TOKEN_ADDRESS]
        }),
      }
    };
    
    we could as add a fees/revenue metric if you have a contract where the fees get sent to

**Comment by llamatester** [comment]
The adapter at projects/DeNet exports TVL: 
        
 
 
    polygon                   0
    polygon-staking           12.12 k
    staking                   12.12 k
    
    total                    0

File: projects/DeNet/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xasai) [reply projects/DeNet/index.js]: Hey, @g1nt0ki, we have applied these changes

**Pull request merged**